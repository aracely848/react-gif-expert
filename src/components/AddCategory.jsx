import { useState } from 'react';
import PropTypes from 'prop-types';

// export const AddCategory = ({setCategories}) => {
export const AddCategory = ({onNewCategory}) => {

    const [ inputValue, setInputValue ] = useState('');

    // const onInputChange = (event) => {
    //     setInputValue(event.target.value)
    // }
    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;

        //setCategories(categories => [ inputValue, ...categories ]);
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <form
            aria-label='form'
            onSubmit={ onSubmit }
            // onSubmit={ (event) => onSubmit(event)}
        >
            <input 
                type="text"  
                placeholder="Buscar Gifs" 
                value={ inputValue }
                onChange = {onInputChange}
                // onChange = { (event) => onInputChange(event) }
            />
        </form>
    )
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
