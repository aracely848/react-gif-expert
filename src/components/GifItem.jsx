import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
    return (
        <div className="card">
            <img src={ url } alt={ title }/>
            <p>{ title }</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
/* TAREA
1. AÃ±adir PropTypes como usamos vite debemos => npm install prop-types
a. title y url obligatorio
2. Evaluar el snapshot
*/
