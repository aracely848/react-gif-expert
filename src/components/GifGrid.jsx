import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const { images, isLoading } = useFetchGifs( category );
 
    return (
        <>
            <h3>{ category }</h3>
            {
                /* FORMA 1
                isLoading
                ? ( <h2>Cargando...</h2> )
                : null*/

                /* FORMA 2 */
                isLoading && ( <h2>Cargando...</h2> )
            }
            <div className="card-grid">
                {
                    // images.map(image => (
                    //     <li key={image.id}>{image.title}</li>
                    // ))
                    images.map(( image ) => (
                        <GifItem 
                            key={ image.id }
                            //Trae todos los datos
                            { ...image } 
                        />
                    ))
                }                
            </div>
        </>
    )
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}

