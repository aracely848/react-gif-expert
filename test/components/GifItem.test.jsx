import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GitIem />', () => { 

    const title = 's01e01 GIF';
    const url = 'https://media4.giphy.com/media/d2PqO9s7M9CNO/giphy.gif?cid=18265621fgbkz39v1x5i743g441bqhoorv3f8zbwzulpjd7y&rid=giphy.gif&ct=g';
    
    test('Debe de hacer match con el snapshot', () => { 
        const { container } = render( <GifItem title={ title } url={ url} />)
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        
        render( <GifItem title={ title } url={ url} />)
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe( url );
        // expect(screen.getByRole('img').alt).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect ( src ).toBe( url );
        expect ( alt ).toBe( title );
    });

    test('Debe de mostrar el titulo en el componente', () => { 
        render( <GifItem title={ title } url={ url} />);
        expect( screen.getByText( title )).toBeTruthy();
    });

});