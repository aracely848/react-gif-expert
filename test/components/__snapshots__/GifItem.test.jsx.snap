// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitIem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="s01e01 GIF"
      src="https://media4.giphy.com/media/d2PqO9s7M9CNO/giphy.gif?cid=18265621fgbkz39v1x5i743g441bqhoorv3f8zbwzulpjd7y&rid=giphy.gif&ct=g"
    />
    <p>
      s01e01 GIF
    </p>
  </div>
</div>
`;
